src/Update/Nix/FetchGit.hs

```diff
@@ -128,48 +72,44 @@ extractFetchGitArgs = \case
--- Massaging the data a little to make it easier to process
+-- Getting updated information from the internet.
 --------------------------------------------------------------------------------
 
--- | The info needed to find the latest git version and the info about where to
--- update the rev and sha256.
-data FetchGitUpdateInfo = FetchGitUpdateInfo{ urlString :: Text
-                                            , revPos    :: SourceSpan
-                                            , sha256Pos :: SourceSpan
-                                            }
-  deriving (Show)
-
--- | Given some arguments to fetchgit, extract the information necessary to
--- update it
-fetchGitArgsToUpdate :: Text -> FetchGitArgs -> Either Warning FetchGitUpdateInfo
-fetchGitArgsToUpdate t as = FetchGitUpdateInfo (extractUrlString (repoLocation as))
-                                           <$> exprSpan t (revExpr as)
-                                           <*> exprSpan t (sha256Expr as)
+getFetchGitLatestInfo :: FetchGitArgs -> IO (Either Warning FetchGitLatestInfo)
+getFetchGitLatestInfo args = do
+  result <- nixPrefetchGit (extractUrlString $ repoLocation args)
+  case result of
+    Left e -> pure $ Left e
+    Right o -> return $ FetchGitLatestInfo args (rev o) (sha256 o)
+                                           <$> (parseISO8601DateToDay (date o))
 
 --------------------------------------------------------------------------------
--- The type we have after calling nix-prefetch-git, A pair of SpanUpdates for
--- each fetchgit call.
+-- Deciding which parts of the Nix file should be updated and how.
 --------------------------------------------------------------------------------
 
--- | A pair of 'SpanUpdate's for updating a single fetchgit value.
-data FetchGitSpanUpdates = FetchGitSpanUpdates{ revUpdate    :: SpanUpdate
-                                              , sha256Update :: SpanUpdate
-                                              }
-  deriving (Show, Data)
-
--- | Create a pair of 'SpanUpdate's given a 'FetchGitUpdateInfo'
-updateInfoToUpdate :: FetchGitUpdateInfo -> IO (Either Warning FetchGitSpanUpdates)
-updateInfoToUpdate ui = do
-  o <- nixPrefetchGit (urlString ui)
-  pure $ prefetchGitOutputToSpanUpdate ui <$> o
-
--- | Use the output of nix-prefetch-git to create a pair of updates
-prefetchGitOutputToSpanUpdate :: FetchGitUpdateInfo
-                              -> NixPrefetchGitOutput
-                              -> FetchGitSpanUpdates
-prefetchGitOutputToSpanUpdate u o =
-  FetchGitSpanUpdates (SpanUpdate (revPos u) (quote $ rev o))
-                      (SpanUpdate (sha256Pos u) (quote $ sha256 o))
-  where quote t = "\"" <> t <> "\""
-
-
+fetchTreeToSpanUpdates :: FetchTree FetchGitLatestInfo -> [SpanUpdate]
+fetchTreeToSpanUpdates (Node maybeVersionExpr cs) =
+  (concatMap fetchTreeToSpanUpdates cs)
+  ++ (toList $ maybeVersionExpr >>= maybeUpdateVersion cs)
```

Comment:
The rest of the code has the operator at the end of the first line when splitting an expression across multiple lines.